samples/extensions/open_cl_interop/open_cl_interop.cpp

```diff
@@ -1,655 +1,905 @@
+	    "cl_khr_device_uuid"};
+	// Platform specific OpenCL extensions for interop
+#if defined(_WIN32)
+	required_extensions.push_back("cl_khr_external_memory_win32");
+	required_extensions.push_back("cl_khr_external_semaphore_win32");
+#else
+	required_extensions.push_back("cl_khr_external_memory_opaque_fd");
+	required_extensions.push_back("cl_khr_external_semaphore_opaque_fd");
+#endif
+
+	// Iterate over all available OpenCL platforms and find the first that fits our requirements (extensions, device UUID)
+
+	cl_uint num_platforms;
+	clGetPlatformIDs_ptr(0, nullptr, &num_platforms);
+
+	std::vector<cl_platform_id> platform_ids(num_platforms);
+	clGetPlatformIDs_ptr(num_platforms, platform_ids.data(), nullptr);
+
+	cl_platform_id selected_platform_id{nullptr};
+	cl_device_id   selected_device_id{nullptr};
+
+	for (auto &platform_id : platform_ids)
+	{
+		cl_uint num_devices;
+		clGetDeviceIDs_ptr(platform_id, CL_DEVICE_TYPE_ALL, 0, nullptr, &num_devices);
+		std::vector<cl_device_id> device_ids(num_devices);
+		clGetDeviceIDs_ptr(platform_id, CL_DEVICE_TYPE_ALL, num_devices, device_ids.data(), nullptr);
+
+		// Check if this platform supports all required extensions
+		size_t extension_string_size = 0;
+		clGetPlatformInfo(platform_id, CL_PLATFORM_EXTENSIONS, 0, nullptr, &extension_string_size);
+
+		std::string extension_string(extension_string_size, ' ');
+		clGetPlatformInfo(platform_id, CL_PLATFORM_EXTENSIONS, extension_string_size, &extension_string[0], nullptr);
+
+		std::vector<std::string> available_extensions{};
+		std::stringstream        extension_stream(extension_string);
+		std::string              extension;
+		while (std::getline(extension_stream, extension, ' '))
+		{
+			// Remove trailing zeroes from all extensions (which otherwise may make support checks fail)
+			extension.erase(std::find(extension.begin(), extension.end(), '\0'), extension.end());
+			available_extensions.push_back(extension);
+		}
+
+		bool extensions_present = true;
+
+		for (auto &extension : required_extensions)
+		{
+			if (std::find(available_extensions.begin(), available_extensions.end(), extension) == available_extensions.end())
+			{
+				extensions_present = false;
+				break;
+			}
+		}
+
+		if (!extensions_present)
+		{
+			continue;
+		}
+
+		// Check every device of this platform and see if it matches our Vulkan device UUID
+		selected_device_id = nullptr;
+		for (auto &device_id : device_ids)
+		{
+			cl_uchar uuid[CL_UUID_SIZE_KHR];
+			clGetDeviceInfo_ptr(device_id, CL_DEVICE_UUID_KHR, sizeof(uuid), &uuid, nullptr);
+
+			bool device_uuid_match = true;
+
+			for (uint32_t i = 0; i < CL_UUID_SIZE_KHR; i++)
+			{
+				if (uuid[i] != physical_device_id_propreties.deviceUUID[i])
+				{
+					device_uuid_match = false;
+					break;
+				}
+			}
+
+			if (!device_uuid_match)
+			{
+				continue;
+			}
+
+			// We found a device with a matching UUID, so use it
+			selected_device_id = device_id;
+			break;
```

Comment:
Maybe:
```
		auto deviceIt = std::find_if(device_ids.begin(),
		                             device_ids.end(),
		                             [&physical_device_id_propreties](auto const &device_id) {
			                             cl_uchar uuid[CL_UUID_SIZE_KHR];
			                             clGetDeviceInfo_ptr(device_id, CL_DEVICE_UUID_KHR, sizeof(uuid), &uuid, nullptr);
			                             return memcmp(uuid, physical_device_id_propreties.deviceUUID, CL_UUID_SIZE_KHR) == 0;
		                             });
```

Reply:
Tbh. I prefer the better readability of a loop ;)

Reply:
As above.